ee/clickhouse/queries/clickhouse_retention.py

```diff
@@ -100,6 +87,56 @@ def calculate_retention(self, filter: Filter, team: Team, total_intervals: int)
 
         return parsed
 
+    def _get_event_filter(self, filter: Filter) -> Tuple[str, str, Dict]:
+        query = ""
+        params: Dict = {}
+
+        first_time_retention = filter.retention_type == RETENTION_FIRST_TIME
+
+        target_entity = (
+            Entity({"id": "$pageview", "type": TREND_FILTER_TYPE_EVENTS})
+            if not filter.target_entity
+            else filter.target_entity
+        )
+
+        returning_entity = (
+            (
+                Entity({"id": "$pageview", "type": TREND_FILTER_TYPE_EVENTS})
+                if not len(filter.entities) > 0
+                else filter.entities[0]
+            )
+            if first_time_retention
+            else target_entity
+        )
+
+        target_query, params = self._get_condition(target_entity)
+        returning_query, returning_params = self._get_condition(
+            returning_entity, "returning" if first_time_retention else ""
+        )
+
+        query = "AND {target_query}".format(target_query=target_query)
+        returning_query = (
+            "AND ({target_query} OR {returning_query})".format(
```

Comment:
I count the first event as part of the return conditions to make sure the first column of events is wrapped in but I think this is based on bad assumptions that people will have clear cut first time events. Going to revisit this tomorrow and readjust this